### PR TITLE
Re-Writing Bans and Notes while Optimizing Banning and Punishments

### PR DESCRIPTION
**Pull request by Geekyhobo**
### Welcome to Hell

- TLDR: Bans and Offenses are being rewritten for more, Clarity, Easy of Use, Reduce of DB clutter, and overall ease of use when it comes to in-game administration
- This is a draft, cause this is gonna take a LOOOT of work and im working rn and will do it bites at a time
- PLEASE help if you have any DB experience cause I done and had a ton of help from Cheif Engineer on how it should be restructured
- Shoutout to Nik and CE for doing some heavy lifting and planning, and for nik and doing the future UI for the banning UI

## Why / Balance

- Currently banning people is pretty tedious and slow this system and rework will allow for bans to be automatically calculated based on what is selected
- The current bans DB structure is, well, scuffed and has probably needed to be killed with fire for a while

## Technical details

- Takes a lot of DB Tables culls their use and uses an Offense instead
- I haven't done the other stuff yet soooooooooooo stand by for more

## Media

Coming Soon

## Breaking changes

**Changelog**

Prob No CL cause no player facing changes

### CODE REVIEW

**Comment by Geekyhobo** [comment]
This should also fix #25204

**Comment by MilonPL** [comment]
@Geekyhobo is this still being worked on?
